src/frontend/screens/AppPasscode/InputPasscode.tsx

```diff
@@ -67,28 +70,40 @@ interface SetPasscodeProps {
   setScreenState: React.Dispatch<React.SetStateAction<PasscodeScreens>>;
 }
 
-/** This screen is used when the user is setting a password, confiriming their password, and will be used when the user is entering their password to enter the app */
+/** This screen is used when the user is setting a password and confiriming their password */
```

Comment:
I think you're maybe trying to do too much with this screen? I don't fully understand what is happening here, but it seems like you are using refs to enable this screen to be used in different places? Perhaps the UI part (without state) can be the part that you re-used, and we just write a different screen  / state logic for each place we use it? I feel like the code seems like it is doing a lot to avoid having 2 or 3 separate components.

Reply:
I can give that a try. The pages are identical, with different text. But I agree that it is a little confusing given that even though the ui is the same, the logic for each page is slightly different.